pyodide_build/cli/skeleton.py

```diff
@@ -48,19 +48,26 @@ def new_recipe_pypi(
     ),
 ) -> None:
     """
-    Create a new package from PyPI.
+    Create a new package recipe from PyPI or update an existing recipe.
     """
 
+    # Determine the recipe directory. If it is specified by the user, we use that;
+    # otherwise, we assume that the recipe directory is the ``packages`` directory
+    # in the root of the Pyodide tree, without the need to initialize the
+    # cross-build environment.
+    #
+    # It is unlikely that a user will run this command outside of the Pyodide
+    # tree, so we do not need to initialize the environment at this stage.
+
+    cwd = Path.cwd()
+    root = build_env.search_pyodide_root(curdir=cwd)
+
+    if not root:
+        root = cwd
```

Comment:
We don't need to calculate these when recipe_dir is specified, so how about putting it in the else block?

Reply:
Makes sense to me, sure! Addressed in 4e099e39d78fb74fb48b883c1df581c51f99410d.